Gravatar/MruImageCache.cs

```diff
@@ -72,19 +82,22 @@ Image IImageCache.GetImage(string imageFileName)
                 throw new ArgumentException(nameof(imageFileName));
             }
 
-            if (_cache.TryGetValue(imageFileName, out var image))
+            lock (_cache)
             {
-                return image;
-            }
+                if (_cache.TryGetValue(imageFileName, out var cachedImage))
+                {
+                    return cachedImage;
+                }
 
-            image = _inner.GetImage(imageFileName);
+                var image = _inner.GetImage(imageFileName);
```

Comment:
⚠️ Typically you want to avoid delegating out during a lock, since you can't *guarantee* completion of the locked code in this case. The method below which uses two locks shows the more desirable pattern.

Reply:
Fixed.